lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -523,18 +523,7 @@ function dosomething_reportback_view_entity($entity) {
     $fid = array_pop($entity->fids);
   }
 
-
-  // @TODO Can we templatize/DRY some of this? It's fairly similiar code in multiple places.
-  $disable_reactions = dosomething_kudos_disable_reactions($entity->nid);
-
-  if (! $disable_reactions) {
-    $kudos = [];
-    $kudos['rbid'] = $entity->rbid;
-    $kudos['disable_reactions'] = $disable_reactions;
-    $kudos['allowed_reactions'] = [dosomething_kudos_get_term_id_by_name('heart')];
-    $kudos['reaction_totals'] = ['heart' => dosomething_kudos_get_total_for_file_by_term_name($entity->rbid, 'heart')];
-    $kudos['existing_kids'] = dosomething_kudos_get_kudos_for_user_by_file($entity->rbid);
-  }
```

Comment:
That's also assuming this is the _only_ function that will need to know if Kudos is disabled. As more functionality is added I'm guessing other things will need to know this, regardless of whether they are trying to grab kudos data or not.